sse: introduce  sync pool for `rawEvent` to optimize memory usage.

liuzengh opened the pull request.

liuzengh committed:
sse: introduce a sync pool for `rawEvent` to optimize memory usage.

liuzengh commented:
@jeevatkm please take a look

jeevatkm reviewed (changes requested):
@liuzengh Can you please check test cases failure?

liuzengh committed:
fix deferred resource cleanup in event parsing

Move the `defer putRawEvent(ed)` call after error handling to ensure resources are properly released only when parsing succeeds. This prevents potential resource leaks when errors occur during event parsing.

liuzengh committed:
Merge remote-tracking branch 'upstream/v3' into event

# Conflicts:
#	sse.go

liuzengh commented:
> @liuzengh Can you please check test cases failure?

fixed

jeevatkm reviewed (approved):
@liuzengh Thanks for fixing test cases.
